core/src/main/java/feast/core/model/EntityV2.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.model;
+
+import com.google.protobuf.Timestamp;
+import feast.core.util.TypeConversion;
+import feast.proto.core.EntityProto;
+import feast.proto.core.EntityProto.*;
+import feast.proto.types.ValueProto.ValueType;
+import java.util.Map;
+import javax.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@javax.persistence.Entity
+@Table(
+    name = "entities_v2",
+    uniqueConstraints = @UniqueConstraint(columnNames = {"name", "project_name"}))
+public class EntityV2 extends AbstractTimestampEntity {
+  @Id @GeneratedValue private long id;
+
+  // Name of the Entity
+  @Column(name = "name", nullable = false)
+  private String name;
+
+  // Project that this Entity belongs to
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "project_name")
+  private Project project;
+
+  // Description of entity
+  @Column(name = "description", columnDefinition = "text")
+  private String description;
+
+  // Columns of entities
+  /** Data type of each entity column: String representation of {@link ValueType} * */
+  private String type;
+
+  // User defined metadata
+  @Column(name = "labels", columnDefinition = "text")
+  private String labels;
+
+  public EntityV2() {
+    super();
+  }
+
+  public EntityV2(
```

Comment:
Would you please add comments to describe the model?

Reply:
Added.